docs/DeveloperGuide.md

```diff
@@ -28,14 +46,23 @@ The `ModuleList` class serves as the main data storage class for the program, an
 The `Module` class serves as a wrapper around a `TaskList`, providing additional attributes including the module code and module description. Within the context of Mod Happy, modules can be viewed as task categories with names, descriptions and other attributes. For this reason, the General Tasks list is implemented as a `Module` under the hood.
 
 > 📔 <span style="color:#00bb00">**NOTE:**</span>
-> 
+>
 > An alternative method of implementing `ModuleList` is shown below, where the default General Tasks
 list is simply represented as a `TaskList` instead of a full-fledged `Module`.
 >
 > ![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/DataAlternative.puml)
-> 
+>
 > While this model is arguably closer to real life, the program logic would have to operate on different object types depending on whether a given `Task` belongs to a user-created Module or the default General Tasks list. This was deemed to increase coupling and introduce too much unnecessary clutter to the code, hence it was not used.
 
+### Storage Component
+**API** : Storage.java
+![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/Ch40gRv1-Mu/tp/branch-A-Storage-puml/docs/Storage.puml)
+The Storage component,
+* Storage interface is implemented by JsonStorage in Mod Happy, which will read and load data to and from json famat.
+* ListStorage can save a ArrayList of any class that extends Object in json format, and read them back into corresponding objects.
+* ModuleListStorage, TaskListStorage inherit from ListStorage, and they manage the write/load of ModuleList and TaskList to/from disk respectively
+* ConfigurationStorage in inherits directly from JsonStorage, and it manage the write/load of Configuration
+* There are navigability to Storage from Main and SaveComand, which handles the load and write data to/from disk respectively.
 ## Implementation
```

Comment:
I remember you mentioned something about how template objects can't be used for one of the functions, and that's why we didn't use `ListStorage<T>` instead of creating 2 classes for ModuleListStorage and TaskListStorage. Maybe we can briefly mention it here in the implementation section. (Just a note for the future)